dandi/download.py

```diff
@@ -236,7 +239,6 @@ def _get_asset_files(
                     )
                     continue
             dandiset = Dandiset(dandiset_path, allow_empty=True)
-            if not dandiset.path_obj.exists():
-                dandiset.path_obj.mkdir()
+            dandiset.path_obj.mkdir(exist_ok=True)  # exist_ok in case of parallel race
```

Comment:
hm... so the download tests we have do not have any directories, sad... oh well -- I tested locally on a large dataset, was all good.